src/nvim/eval.lua

```diff
@@ -17,9 +17,10 @@
 --- @field deprecated? true
 --- @field returns? string|false
 --- @field returns_desc? string
+--- @field generics string[] Used to write `---@generics` annotations over a function.
 --- @field signature? string
 --- @field desc? string
---- @field params [string, string, string][]
+--- @field params {[1]: string, [2]: string}[] The input of any `signature`.
```

Comment:
This looks wrong. The old format was a 3 string tuple of `name, type, desc`.

Reply:
I forgot to leave a note about this, thank you for reminding me - the original `[string, string, string][]` is incorrect.

This can be verified with `lua-language-server`

`lua-language-sever --check repro.lua`
`repro.lua`
```lua
---@class Foo
---@field [string, string string][]
```

`Diagnosis complete, 1 problems found, see ~/path/to/lua-language-server/libexec/log/check.json`

```json
{
    "file:///home/selecaoone/temp/annotation_key_test.lua": [
        {
            "code": "luadoc-miss-symbol",
            "data": "syntax",
            "message": "`]` expected.",
            "range": {
                "end": {
                    "character": 17,
                    "line": 1
                },
                "start": {
                    "character": 17,
                    "line": 1
                }
            },
            "severity": 2,
            "source": "Lua Syntax Check."
        }
    ]
}
```

`[string, string, string?][]` has the same problem.

Also if you run `:InspectTree`, the parsed tree contains ERROR on that line.

That said though my suggestion doesn't appear to work 100% either. I think it's a bug with LuaLS

```lua
---@class Foo
---@field [1] string

---@type Foo
local foo = {[1]="asdf"}

foo[2]:match("asdfasfd") -- This should error, but doesn't


---@class Bar
---@field thing string

---@type Bar
local foo = {thing="asdf"}

foo[2]:match("asdfasfd") -- This should error, but doesn't
foo.does_not_exist:match("asdfasfd") -- This should error, and it does
foo["does_not_exist"]:match("asdfasfd") -- This should error, but doesn't
```

It seems that accessing undefined numbered fields does not error. So either my suggestion is wrong and there's a better way to define it or it's a bug with LuaLS. Not sure. But the previous syntax does cause an error

Reply:
> Also if you run :InspectTree, the parsed tree contains ERROR on that line.

The output of InspectTree doesn't matter for Lua annotations. What matters is how LuaLS processes them.

Reply:
![image](https://github.com/user-attachments/assets/e7fec587-3cbe-4943-a9af-d82fb03ff358)

It errors for me both in-editor and using the `--check` terminal flag. With either `---@field params [string, string, string][]`  or `---@field params [string, string, string?][]`. What do you see when you run lua-language-server?

Reply:
Tuples were added in 3.8.0 https://github.com/LuaLS/lua-language-server/pull/2505.

Newest version is 3.13

Reply:
You're right, I'll revert this change.

On an unrelated note, it looks like 3.8.0+ has broken `lua-language-server --check`. But since that issue is unrelated to this PR, I'll make a separate issue over in their repo

Thank you for explaining it to me!

Reply:
--check works for me on 3.13. There might be some changes in how it works though.

Reply:
Interesting, I can't get it working on two machines so I'm not sure. Hopefully it's something simple.

About this PR, I reverted the change fully. If you want me to add `[string, string, string?][]` instead or something I can add that in.

Reply:
That or what it was before.